### DISCUSSION TITLE
Universal Sentence Encoder Failing to Deploy on SageMaker Notebook instance]

### DISCUSSION THREAD

**Discussion by singularity014** [aws/sagemaker-python-sdk#2376] [Help]
Hello I am trying to use TensorFlow Serving for using Universal Sentence Encoder using 
Sagemaker's NoteBook instance. 
It throws the error - 
" The primary container for production variant AllTraffic did not pass the ping health check. Please check CloudWatch logs for this endpoint..! "
""

**To reproduce**
A clear, step-by-step set of instructions to reproduce the bug.

from sagemaker import get_execution_role
import sagemaker
role = get_execution_role()
from sagemaker.tensorflow.serving import Model, Predictor

sagemaker_model = Model(
    model_data = 's3://naister-platform-models/MUSE_TAR/universal-sentence-encoder.tar.gz',
    role = role
)

**predictor = sagemaker_model.deploy(
    initial_instance_count=1, 
    instance_type='ml.t2.medium', 
    endpoint_name="MuseEndpointLg"
)**

input = ['I am a good guy.', 'I am a bad guy, very bad.']
output = predictor.predict(input)

**Expected behavior**
Similarity score between the two input sentences.

**Screenshots or logs**
![image](https://user-images.githubusercontent.com/49594316/86568767-aa6f9000-bf6d-11ea-8831-24643ddcdb59.png)

" The primary container for production variant AllTraffic did not pass the ping health check. Please check CloudWatch logs for this endpoint..! "
""

**Comment by metrizable** [comment]
@singularity014 Thanks for filing the issue. I have a few questions:

* Have you checked the CloudWatch logs for the endpoint? And, if so, can you share?
* I see you are using an `ml.t2.medium`. Is your tar.gz of reasonable size?
* From [here](https://docs.aws.amazon.com/sagemaker/latest/dg/your-algorithms-inference-code.html#your-algorithms-inference-algo-ping-requests), your endpoint has 4 minutes to move to healthy state "by consistently responding with 200s". Is it possible that start-up is taking longer than this? The logs may provide some insight. Trying a different instance type may yield results, too.

**Comment by singularity014** [comment]
@metrizable thanks for addressing the question.

- I am getting following error log 'No saved model found'

![image](https://user-images.githubusercontent.com/49594316/86649762-53f06900-bfe2-11ea-8152-b6cc129c56a5.png).

- The tar.gz model is of 318 MB in size.

- Yes, the enpoint creation process was taking very long :
**predictor = sagemaker_model.deploy(
initial_instance_count=1,
instance_type='ml.t2.medium',
endpoint_name="MuseEndpointLg"
)**

It took almost 15 minutes to run this , before it threw the error.

**Comment by laurenyu** [comment]
sorry for the delayed response here. what are the contents of `s3://naister-platform-models/MUSE_TAR/universal-sentence-encoder.tar.gz`?
